pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -255,11 +266,35 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	if node.Spec.ProviderID == "" {
 		return fmt.Errorf("node %s doesn't have providerID", nodeName)
 	}
-
-	cidrStrings, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
+	// fetch gce instance
+	instance, err := ca.cloud.InstanceByProviderID(node.Spec.ProviderID)
 	if err != nil {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
-		return fmt.Errorf("failed to get cidr(s) from provider: %v", err)
+		return fmt.Errorf("failed to get instance from provider: %v", err)
+	}
+
+	cidrStrings := make([]string, 0)
+	var northInterfaces networkv1.NorthInterfacesAnnotation
+	var additionalNodeNetworks networkv1.MultiNetworkAnnotation
+
+	if len(instance.NetworkInterfaces) == 0 || (len(instance.NetworkInterfaces) == 1 && len(instance.NetworkInterfaces[0].AliasIpRanges) == 0) {
+		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
+		return fmt.Errorf("failed to allocate cidr: Node %v has no ranges from which CIDRs can be allocated", node.Name)
+	}
+	// nodes in clusters WITHOUT multi-networking are expected to have only 1 network-interface with 1 alias IP range.
+	if len(instance.NetworkInterfaces) == 1 && len(instance.NetworkInterfaces[0].AliasIpRanges) == 1 {
```

Comment:
Probably an irrelevant question - do we need to take care of route-based cluster, which doesn't have alias ranges?

Reply:
No. We cannot have multiple interfaces for routes based clusters.